CI broken on main: Deploy to staging

## CI failure on main branch

**Run**: https://github.com/alexsiri7/reli/actions/runs/24610136348
**Failed jobs**: Deploy to staging

The latest CI run on main has failed. This needs to be diagnosed and fixed.

Steps:
1. Check the failed run: `gh run view 24610136348 --log-failed`
2. Identify the root cause
3. Fix and push to main

## 🔍 Investigation: CI broken on main: Deploy to staging

**Type**: `BUG`

### Assessment

| Metric | Value | Reasoning |
|--------|-------|-----------|
| Severity | `HIGH` | Staging and production deploys are fully blocked on every CI run |
| Complexity | `LOW` | Single file change — replace two `appleboy/ssh-action@v1` blocks with native SSH |
| Confidence | `HIGH` | Error log is unambiguous: 504 on drone-ssh binary download |

---

### Problem Statement

`appleboy/ssh-action@v1` downloads the `drone-ssh` binary from GitHub releases at runtime. This download is failing with HTTP 504 gateway timeouts, causing the deploy step to abort before any SSH connection is made. Staging and production deploys are both blocked.

---

### Root Cause Analysis

WHY: "Deploy staging via SSH" exits with code 4
↓ BECAUSE: `appleboy/ssh-action@v1` tries to download `drone-ssh` at startup
↓ BECAUSE: `https://github.com/appleboy/drone-ssh/releases/download/v1.8.2` returns 504 six times in a row

```
Downloading drone-ssh-1.8.2-linux-amd64 from https://github.com/appleboy/drone-ssh/releases/download/v1.8.2
curl: (22) The requested URL returned error: 504
[...5 more retries...]
Failed to download drone-ssh-1.8.2-linux-amd64
##[error]Process completed with exit code 4.
```

→ **ROOT CAUSE**: `appleboy/ssh-action@v1` downloads its own binary at runtime rather than bundling it. Fix: replace with native `ssh` (bundled on Ubuntu runners — no download required).

---

### Implementation Plan

| Step | File | Change |
|------|------|--------|
| 1 | `.github/workflows/staging-pipeline.yml:37-59` | Replace staging `appleboy/ssh-action@v1` with setup-ssh-key + native `ssh` |
| 2 | `.github/workflows/staging-pipeline.yml:118-154` | Replace production `appleboy/ssh-action@v1` with setup-ssh-key + native `ssh` |

<details>
<summary>📋 Detailed Implementation Steps</summary>

**For each deploy step** (staging at line 37, production at line 120), replace:

```yaml
- name: Deploy via SSH
  uses: appleboy/ssh-action@v1
  env:
    GHCR_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    GHCR_USER: ${{ github.actor }}
    DEPLOY_SHA: ${{ steps.sha.outputs.sha }}
  with:
    host: 100.120.193.82
    username: ${{ secrets.DEPLOY_USER }}
    key: ${{ secrets.SSH_DEPLOY_KEY }}
    envs: DEPLOY_SHA,GHCR_TOKEN,GHCR_USER
    script: |
      # ... deploy script ...
```

With:

```yaml
- name: Setup SSH key
  run: |
    mkdir -p ~/.ssh
    printf '%s' "${{ secrets.SSH_DEPLOY_KEY }}" > ~/.ssh/deploy_key
    chmod 600 ~/.ssh/deploy_key
    ssh-keyscan -H 100.120.193.82 >> ~/.ssh/known_hosts

- name: Deploy via SSH
  env:
    GHCR_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    GHCR_USER: ${{ github.actor }}
    DEPLOY_SHA: ${{ steps.sha.outputs.sha }}
  run: |
    ssh -i ~/.ssh/deploy_key ${{ secrets.DEPLOY_USER }}@100.120.193.82 << EOF
    export GHCR_TOKEN="${GHCR_TOKEN}"
    export GHCR_USER="${GHCR_USER}"
    export DEPLOY_SHA="${DEPLOY_SHA}"
    # ... same deploy script with \$VAR escaped for remote eval ...
    EOF
```

**Key escaping rule**: Inside the heredoc, variables evaluated *locally* (before SSH) use `$VAR`; variables evaluated *on the remote host* use `\$VAR`.

</details>

---

### Validation

```bash
gh workflow run staging-pipeline.yml --repo alexsiri7/reli
# Verify "Deploy staging via SSH" step has no "Downloading drone-ssh" line
```

---

### Next Step

To implement: `/implement-issue 650`

---
*Investigated by Claude • 2026-04-18T18:00:00Z*

--- a/.github/workflows/staging-pipeline.yml
+++ b/.github/workflows/staging-pipeline.yml
@@ -37,9 +37,13 @@ jobs:
       - name: Setup SSH key
         run: |
           mkdir -p ~/.ssh
-          printf '%s' "${{ secrets.SSH_DEPLOY_KEY }}" > ~/.ssh/deploy_key
+          printf '%s\n' "${{ secrets.SSH_DEPLOY_KEY }}" > ~/.ssh/deploy_key
           chmod 600 ~/.ssh/deploy_key
           ssh-keyscan -H 100.120.193.82 >> ~/.ssh/known_hosts
+          if ! grep -q "100.120.193.82" ~/.ssh/known_hosts; then
+            echo "ERROR: ssh-keyscan produced no output — host may be unreachable"
+            exit 1
+          fi
 
       - name: Deploy staging via SSH
         env:
@@ -47,7 +51,7 @@ jobs:
           GHCR_USER: ${{ github.actor }}
           DEPLOY_SHA: ${{ steps.sha.outputs.sha }}
         run: |
-          ssh -i ~/.ssh/deploy_key ${{ secrets.DEPLOY_USER }}@100.120.193.82 << EOF
+          ssh -o BatchMode=yes -o ConnectTimeout=30 -i ~/.ssh/deploy_key ${{ secrets.DEPLOY_USER }}@100.120.193.82 << EOF
           export GHCR_TOKEN="${GHCR_TOKEN}"
           export GHCR_USER="${GHCR_USER}"
           export DEPLOY_SHA="${DEPLOY_SHA}"
@@ -133,9 +137,13 @@ jobs:
       - name: Setup SSH key
         run: |
           mkdir -p ~/.ssh
-          printf '%s' "${{ secrets.SSH_DEPLOY_KEY }}" > ~/.ssh/deploy_key
+          printf '%s\n' "${{ secrets.SSH_DEPLOY_KEY }}" > ~/.ssh/deploy_key
           chmod 600 ~/.ssh/deploy_key
           ssh-keyscan -H 100.120.193.82 >> ~/.ssh/known_hosts
+          if ! grep -q "100.120.193.82" ~/.ssh/known_hosts; then
+            echo "ERROR: ssh-keyscan produced no output — host may be unreachable"
+            exit 1
+          fi
 
       - name: Deploy via SSH
         env:
@@ -143,7 +151,7 @@ jobs:
           GHCR_USER: ${{ github.actor }}
           DEPLOY_SHA: ${{ needs.deploy-staging.outputs.deploy_sha }}
         run: |
-          ssh -i ~/.ssh/deploy_key ${{ secrets.DEPLOY_USER }}@100.120.193.82 << EOF
+          ssh -o BatchMode=yes -o ConnectTimeout=30 -i ~/.ssh/deploy_key ${{ secrets.DEPLOY_USER }}@100.120.193.82 << EOF
           export GHCR_TOKEN="${GHCR_TOKEN}"
           export GHCR_USER="${GHCR_USER}"
           export DEPLOY_SHA="${DEPLOY_SHA}"

--- a/.github/workflows/staging-pipeline.yml
+++ b/.github/workflows/staging-pipeline.yml
@@ -34,30 +34,36 @@ jobs:
             echo "sha=${{ github.event.workflow_run.head_sha }}" >> "$GITHUB_OUTPUT"
           fi
 
+      - name: Setup SSH key
+        run: |
+          mkdir -p ~/.ssh
+          printf '%s' "${{ secrets.SSH_DEPLOY_KEY }}" > ~/.ssh/deploy_key
+          chmod 600 ~/.ssh/deploy_key
+          ssh-keyscan -H 100.120.193.82 >> ~/.ssh/known_hosts
+
       - name: Deploy staging via SSH
-        uses: appleboy/ssh-action@v1
         env:
           GHCR_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           GHCR_USER: ${{ github.actor }}
           DEPLOY_SHA: ${{ steps.sha.outputs.sha }}
-        with:
-          host: 100.120.193.82
-          username: ${{ secrets.DEPLOY_USER }}
-          key: ${{ secrets.SSH_DEPLOY_KEY }}
-          envs: DEPLOY_SHA,GHCR_TOKEN,GHCR_USER
-          script: |
-            IMAGE="ghcr.io/alexsiri7/reli"
+        run: |
+          ssh -i ~/.ssh/deploy_key ${{ secrets.DEPLOY_USER }}@100.120.193.82 << EOF
+          export GHCR_TOKEN="${GHCR_TOKEN}"
+          export GHCR_USER="${GHCR_USER}"
+          export DEPLOY_SHA="${DEPLOY_SHA}"
+          IMAGE="ghcr.io/alexsiri7/reli"
 
-            echo "$GHCR_TOKEN" | docker login ghcr.io -u "$GHCR_USER" --password-stdin
-            docker pull "$IMAGE:$DEPLOY_SHA"
+          echo "\$GHCR_TOKEN" | docker login ghcr.io -u "\$GHCR_USER" --password-stdin
+          docker pull "\$IMAGE:\$DEPLOY_SHA"
 
-            cd /mnt/ext-fast/gc/rigs/reli
-            # fetch+reset instead of pull --rebase: handles server-side unstaged changes cleanly
-            # Note: resets tracked files only; untracked files (.env, etc.) are preserved
-            git fetch origin main
-            git reset --hard origin/main
-            docker rm -f reli-staging 2>/dev/null || true
-            RELI_IMAGE_TAG="$DEPLOY_SHA" docker compose -f docker-compose.staging.yml up -d --remove-orphans
+          cd /mnt/ext-fast/gc/rigs/reli
+          # fetch+reset instead of pull --rebase: handles server-side unstaged changes cleanly
+          # Note: resets tracked files only; untracked files (.env, etc.) are preserved
+          git fetch origin main
+          git reset --hard origin/main
+          docker rm -f reli-staging 2>/dev/null || true
+          RELI_IMAGE_TAG="\$DEPLOY_SHA" docker compose -f docker-compose.staging.yml up -d --remove-orphans
+          EOF
 
       - name: Wait for staging health
         run: |
@@ -124,62 +130,68 @@ jobs:
         with:
           authkey: ${{ secrets.TAILSCALE_AUTHKEY }}
 
+      - name: Setup SSH key
+        run: |
+          mkdir -p ~/.ssh
+          printf '%s' "${{ secrets.SSH_DEPLOY_KEY }}" > ~/.ssh/deploy_key
+          chmod 600 ~/.ssh/deploy_key
+          ssh-keyscan -H 100.120.193.82 >> ~/.ssh/known_hosts
+
       - name: Deploy via SSH
-        uses: appleboy/ssh-action@v1
         env:
           GHCR_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           GHCR_USER: ${{ github.actor }}
           DEPLOY_SHA: ${{ needs.deploy-staging.outputs.deploy_sha }}
-        with:
-          host: 100.120.193.82
-          username: ${{ secrets.DEPLOY_USER }}
-          key: ${{ secrets.SSH_DEPLOY_KEY }}
-          envs: DEPLOY_SHA,GHCR_TOKEN,GHCR_USER
-          script: |
-            IMAGE="ghcr.io/alexsiri7/reli"
+        run: |
+          ssh -i ~/.ssh/deploy_key ${{ secrets.DEPLOY_USER }}@100.120.193.82 << EOF
+          export GHCR_TOKEN="${GHCR_TOKEN}"
+          export GHCR_USER="${GHCR_USER}"
+          export DEPLOY_SHA="${DEPLOY_SHA}"
+          IMAGE="ghcr.io/alexsiri7/reli"
 
-            echo "$GHCR_TOKEN" | docker login ghcr.io -u "$GHCR_USER" --password-stdin
+          echo "\$GHCR_TOKEN" | docker login ghcr.io -u "\$GHCR_USER" --password-stdin
 
-            # Tag current running image as :rollback before deploying
-            CURRENT_ID=$(docker inspect --format='{{.Image}}' reli-reli-1 2>/dev/null || true)
-            if [ -n "$CURRENT_ID" ]; then
-              docker tag "$CURRENT_ID" "$IMAGE:rollback"
+          # Tag current running image as :rollback before deploying
+          CURRENT_ID=\$(docker inspect --format='{{.Image}}' reli-reli-1 2>/dev/null || true)
+          if [ -n "\$CURRENT_ID" ]; then
+            docker tag "\$CURRENT_ID" "\$IMAGE:rollback"
+          fi
+
+          # Pull the new image (should already be cached from staging)
+          docker pull "\$IMAGE:\$DEPLOY_SHA"
+
+          # Update and restart production
+          cd /mnt/ext-fast/gc/rigs/reli
+          git pull --rebase
+          RELI_IMAGE_TAG="\$DEPLOY_SHA" docker compose up -d
+
+          # Verify the container is healthy
+          echo "Waiting for health check..."
+          HEALTHY=false
+          for i in 1 2 3 4 5; do
+            sleep 3
+            if curl -sf --max-time 5 http://localhost:8000/healthz | grep -q '"status":"ok"'; then
+              HEALTHY=true
+              break
             fi
+            echo "Health check attempt \$i failed, retrying..."
+          done
 
-            # Pull the new image (should already be cached from staging)
-            docker pull "$IMAGE:$DEPLOY_SHA"
-
-            # Update and restart production
-            cd /mnt/ext-fast/gc/rigs/reli
-            git pull --rebase
-            RELI_IMAGE_TAG="$DEPLOY_SHA" docker compose up -d
-
-            # Verify the container is healthy
-            echo "Waiting for health check..."
-            HEALTHY=false
-            for i in 1 2 3 4 5; do
-              sleep 3
-              if curl -sf --max-time 5 http://localhost:8000/healthz | grep -q '"status":"ok"'; then
-                HEALTHY=true
-                break
-              fi
-              echo "Health check attempt $i failed, retrying..."
-            done
-
-            if [ "$HEALTHY" = true ]; then
-              echo "Deploy successful: $DEPLOY_SHA (health check passed)"
+          if [ "\$HEALTHY" = true ]; then
+            echo "Deploy successful: \$DEPLOY_SHA (health check passed)"
+          else
+            echo "Deploy FAILED — health check did not pass after 5 attempts"
+            echo "Rolling back to previous image..."
+            RELI_IMAGE_TAG=rollback docker compose up -d
+            sleep 5
+            if curl -sf --max-time 5 http://localhost:8000/healthz | grep -q '"status":"ok"'; then
+              echo "Rollback successful — previous version is healthy"
             else
-              echo "Deploy FAILED — health check did not pass after 5 attempts"
-              echo "Rolling back to previous image..."
-              RELI_IMAGE_TAG=rollback docker compose up -d
-              sleep 5
-              if curl -sf --max-time 5 http://localhost:8000/healthz | grep -q '"status":"ok"'; then
-                echo "Rollback successful — previous version is healthy"
-              else
-                echo "WARNING: Rollback version also failed health check"
-              fi
-              exit 1
+              echo "WARNING: Rollback version also failed health check"
             fi
+            exit 1
+          fi
+          EOF
 
       - name: Notify prod landing
         if: success()

--- a/frontend/src/components/Sidebar.tsx
+++ b/frontend/src/components/Sidebar.tsx
@@ -571,10 +571,13 @@ export function Sidebar() {
 
   useEffect(() => {
     const mql = window.matchMedia('(min-width: 768px)')
-    const handler = (e: MediaQueryListEvent) => setIsOpen(e.matches)
+    const handler = (e: MediaQueryListEvent) => {
+      setIsOpenLocal(e.matches)
+      setSidebarOpen(e.matches)
+    }
     mql.addEventListener('change', handler)
     return () => mql.removeEventListener('change', handler)
-  }, [])
+  }, [setSidebarOpen])
 
   const upcoming = things
     .filter(t => t.checkin_date != null)

